### DISCUSSION TITLE
Allowlist a domain for `lint/a11y/noBlankTarget`

### DISCUSSION THREAD

**Discussion by GRA0007** [biomejs/biome#3513] [Rule suggestion]
We have a few sites that we manage, and currently any links between them we manually disable the `lint/a11y/noBlankTarget` rule so our analytics can track clicks between them. We're not worried about security given that all the sites are ours.

This results in about 100 or so links with a manual ignore comment. I'd like to propose a config option that allowlists certain domains to allow `target="_blank"` without `noreferrer` set.

Something like this would be great:

```json
{
  "noBlankTarget": {
    "options": {
      "allowDomains": ["ourdomain.com", "oursubdomain.hostingprovider.net"]
    }
  }
}
```

I believe this aligns with the ethos of this rule, as it's designed to prevent security issues with 3rd party sites, but managing multiple sites and linking between them is a common use-case.

**Comment by dyc3** [comment]
Seems reasonable to me. Should be reasonably easy for a new contributor, could be a new `good first issue`.
